ambari-server/src/main/java/org/apache/ambari/server/controller/internal/ServiceGroupResourceProvider.java

```diff
@@ -422,6 +436,11 @@ private void validateCreateRequests(Set<ServiceGroupRequest> requests, Clusters
       }
       serviceGroupNames.get(clusterName).add(serviceGroupName);
 
+      if (request.getMpackNames().size() != 1) {
+        String errmsg = "Invalid arguments, " + request.getMpackNames().size() + " mpack(s) found in the service group " + serviceGroupName + ", only one mpack is allowed";
```

Comment:
Use String.format() here.

Reply:
It seems that concatenation has better performance than string.format().

Reply:
For messages like this, we're more interested in readability in code than performance.  We've been using String.format() in more and more places to be a bit more consistent with logging syntax (though not identical).

Reply:
Ok, I will update it.

Reply:
Should I use StringBuilder instead? I saw your another comment mentioned it.